### PR TITLE
Made NoteRepository singleton

### PR DESCRIPTION
**Pull request by vladyslav-babych**
1. Removed redundant imports
2. Edited `NoteDatabase` class
3. Made `NoteRepository` singleton

### CODE REVIEW

File: app/src/main/java/com/otaman/notes/viewmodel/EditNoteViewModel.kt [hunk 1]
```diff
@@ -10,14 +10,14 @@ import kotlinx.coroutines.Dispatchers
...
 
 class EditNoteViewModel(application: Application): AndroidViewModel(application) {
-    private val dao = NoteDatabase.getDatabase(application).getNoteDao()
-    private val repository: NoteRepository = NoteRepository(dao)
+    private val dao = NoteDatabase.getInstance(application).getNoteDao()
+    private val repository = NoteRepository.getInstance(dao)
 
     fun updateNote(note: Note) = viewModelScope.launch(Dispatchers.IO) {
-        repository.updateNote(note)
+        repository!!.updateNote(note)
```
Comment (RuslanKharkevych) [inline app/src/main/java/com/otaman/notes/viewmodel/EditNoteViewModel.kt]: `repository` should not be nullable

File: app/src/main/java/com/otaman/notes/model/room/NoteDatabase.kt [hunk 2]
```diff
@@ -13,16 +13,16 @@ abstract class NoteDatabase: RoomDatabase() {
...
-                val instance = Room.databaseBuilder(
-                    context.applicationContext,
-                    NoteDatabase::class.java,
-                    "note_database"
-                ).build()
-                INSTANCE = instance
-                instance
+        fun getInstance(context: Context): NoteDatabase =
+            INSTANCE ?: synchronized(this) {
+                INSTANCE ?: buildDatabase(context).also { INSTANCE = it }
```
Comment (RuslanKharkevych) [inline app/src/main/java/com/otaman/notes/model/room/NoteDatabase.kt]: `INSTANCE` is always null on line 18. Actually, I don't see any issues with `NoteDatabase` class before changes

**Review by RuslanKharkevych — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/otaman/notes/viewmodel/EditNoteViewModel.kt (hunk 1 above)
Comment (RuslanKharkevych) [reply app/src/main/java/com/otaman/notes/viewmodel/EditNoteViewModel.kt]: You removed the force-wrapping operator, but repository is still nullable

Comment (vladyslav-babych) [reply app/src/main/java/com/otaman/notes/viewmodel/EditNoteViewModel.kt]: Probably fixed

**Review by RuslanKharkevych — APPROVED** [review]

**Pull request merged**